utils/modularize.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @author mrdoob / http://mrdoob.com/
+ */
+
+var fs = require( 'fs' );
+
+var srcFolder = '../examples/js/';
+var dstFolder = '../examples/jsm/';
+
+var files = [
+	{ path: 'controls/OrbitControls.js', ignoreList: [] },
+	{ path: 'controls/MapControls.js', ignoreList: [] },
+	{ path: 'controls/TrackballControls.js', ignoreList: [] },
+	// { file: 'controls/TransformControls.js', ignoreList: [] },
+	{ path: 'loaders/GLTFLoader.js', ignoreList: [ 'NoSide', 'Matrix2', 'DDSLoader', 'DRACOLoader', 'BufferGeometryUtils' ] },
+	{ path: 'loaders/OBJLoader.js', ignoreList: [] }
+];
+
+for ( var i = 0; i < files.length; i ++ ) {
+
+	var file = files[ i ];
+	convert( file.path, file.ignoreList );
+
+}
+
+//
+
+function convert( path, ignoreList ) {
+
+	var contents = fs.readFileSync( srcFolder + path, 'utf8' );
+
+	var className = '';
+	var dependencies = {};
+
+	// class name
+
+	contents = contents.replace( /THREE\.([a-zA-Z0-9]+) = /g, function ( match, p1 ) {
+
+		className = p1;
+
+		console.log( className );
+
+		return `_IMPORTS_\n\nvar ${p1} = `;
```

Comment:
Why `var` not `const`?

Reply:
No preference.

Reply:
> sompylasar 3 hours ago
> Why var not const?

> mrdoob an hour ago
> No preference.

🤔 How so? Too shy to ask, but... are you aware of the difference, like the function scope versus block scope?

For example, in the code above, the `i` and the `file` variables leak out at least into the file-level function scope, even though they are written within the `for` loop block scope.
```js
for ( var i = 0; i < files.length; i ++ ) {

	var file = files[ i ];
```

Reply:
> How so? Too shy to ask, but... are you aware of the difference, like the function scope versus block scope?

Yes, I'm aware.

Reply:
Next step is to investigate ESM to UMD or the vice versa. Using `var` or `const` (obviously `const`) and similar details is something I would leave for later.

Reply:
Certainly let/const are better, yeah... it's just that we don't use let/const elsewhere, and inconsistency with the rest of the library is the tradeoff here.

+1 from me for moving the library to let/const along with other ES features, but no preference on whether this script is the first instance of that.

Reply:
Ok that's fair. This decision contributes to the strategy. If we had a document, this decision would be written down there, once and for everyone, until further notice.